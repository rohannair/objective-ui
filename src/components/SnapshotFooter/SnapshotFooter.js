import React, { PropTypes } from 'react'
import styled from 'styled-components'
import gql from 'graphql-tag'

import Reaction from '../Reaction'

const SnapshotFooter = styled(({
  className,
  count,
  toggleAction,
  isLiked,
  readOnly
}) => {
  // if (!count) return <div className={className} />
  return (
    <div className={className}>
      <Reaction
        isLiked={isLiked}
        onClick={toggleAction}
        readOnly={readOnly}
       />
       <span style={{
         fontSize: '0.85em',
         color: '#f36',
         fontWeight: 'bold'
       }}>{ count }</span>
    </div>
  )
})`
  display: flex;
  flex-direction: row;
  border-top: 1px dashed #ddd;
  padding: 15px 10px 0;
  align-items: center;
`

SnapshotFooter.mutations = {
  addReaction: gql`
    mutation addReaction($reactionId: Int!, $snapshotId: Int!) {
      addReaction(reactionId: $reactionId, snapshotId: $snapshotId) {
        id
        ...ReactionBoxFragment
      }
    }
    ${Reaction.fragments.reaction}
  `,

  deleteReaction: gql`
    mutation deleteReaction($reactionId: Int!, $snapshotId: Int!) {
      deleteReaction(reactionId: $reactionId, snapshotId: $snapshotId) {
        id
      }
    }
  `
}

SnapshotFooter.fragments = {
  footer: gql`
    fragment SnapshotFooterFragment on Snapshot {
      reactions {
        id
        ...ReactionBoxFragment
      }
    }
    ${Reaction.fragments.reaction}
  `
}

export default SnapshotFooter
