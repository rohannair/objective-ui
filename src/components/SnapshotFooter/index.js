import SnapshotFooter from './SnapshotFooter'
export default SnapshotFooter
