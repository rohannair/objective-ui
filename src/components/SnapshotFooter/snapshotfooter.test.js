import React from 'react'

import SnapshotFooter from './index.js'

describe('Test suite for SnapshotFooter component', () => {
  it('SnapshotFooter should exist', () => {
    let wrapper = shallow(<SnapshotFooter/>)
    expect(wrapper).toMatchSnapshot()
  })
})

