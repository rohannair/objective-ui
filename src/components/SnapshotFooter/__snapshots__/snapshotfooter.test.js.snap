exports[`Test suite for SnapshotFooter component SnapshotFooter should exist 1`] = `
<Component
  className="jZmhLl" />
`;
